qiskit/algorithms/amplitude_amplifiers/grover.py

```diff
@@ -246,7 +251,16 @@ def amplify(self, amplification_problem: AmplificationProblem) -> "GroverResult"
                 )
 
             all_circuit_results.append(circuit_results)
-            oracle_evaluation = amplification_problem.is_good_state(top_measurement)
+
+            # only check if top measurement is a good state if an is_good_state arg has been provided
+            if amplification_problem.is_good_state is not None:
+                oracle_evaluation = amplification_problem.is_good_state(top_measurement)
```

Comment:
I think `oracle_evaluation` should be `None` (instead of `False`) if `is_good_state` is not set, because we cannot check it at all.